### PR TITLE
Fix git version and pin some others.

### PR DESCRIPTION
**Pull request by ktlim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -149,10 +149,10 @@ outputs:
         - idds-client >=2.0.9
         - idds-doma >=2.0.9
         - iminuit >=2.24.0
-        - lmfit >=1.2.2
+        - lmfit >=1.2.2,<1.3
         - lsstdesc.coord >=1.3.0
         - lsst-ts-xml >=20.1.0
-        - matplotlib-base >=3.8.2
+        - matplotlib-base >=3.8.2,<3.9
```
Comment (erykoff) [inline recipe/meta.yaml:155]: Why is this needed? (Skyproj at least is compatible)

Comment (ktlim) [reply recipe/meta.yaml:155]: I took it from the `main` branch; I think it was for "possible API breakage" reasons.  I can try 3.9(.1), but I doubt we have good test coverage.

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender
